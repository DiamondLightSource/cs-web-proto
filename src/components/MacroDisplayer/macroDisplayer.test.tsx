import React from "react";
import { MacroDisplayer } from "./macroDisplayer";
import { configure, mount, ReactWrapper } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { create, ReactTestRenderer } from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();

let snapshot: ReactTestRenderer;
let wrapper: ReactWrapper;

beforeEach((): void => {
  const store = mockStore({
    macroMap: {
      a: "b"
    }
  });
  wrapper = mount(
    <Provider store={store}>
      <MacroDisplayer />
    </Provider>
  );
  snapshot = create(
    <Provider store={store}>
      <MacroDisplayer />
    </Provider>
  );
});

describe("<Input />", (): void => {
  test("it matches the snapshot", (): void => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
  test("it renders a table", (): void => {
    expect(wrapper.find("table").length).toEqual(1);
  });
});
