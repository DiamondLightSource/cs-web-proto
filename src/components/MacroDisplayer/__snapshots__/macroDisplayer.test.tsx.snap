// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> it matches the snapshot 1`] = `
<table
  style={
    Object {
      "borderCollapse": "collapse",
      "display": "block",
      "position": "relative",
      "tableLayout": "fixed",
      "width": "100%",
    }
  }
>
  <tbody
    style={
      Object {
        "display": "table",
        "width": "100%",
      }
    }
  >
    <tr>
      <th
        style={
          Object {
            "width": "50%",
          }
        }
      >
        Macro name
      </th>
      <th
        style={
          Object {
            "width": "50%",
          }
        }
      >
        Macro value
      </th>
    </tr>
    <tr>
      <td>
        a
      </td>
      <td>
        b
      </td>
    </tr>
  </tbody>
</table>
`;
